### PR TITLE
fix: Graph chart colors

### PR DESCRIPTION
**Pull request by michael-s-molina**
### SUMMARY
Fixes the Graph chart to obey to the color scheme when there are no color categories.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
https://github.com/user-attachments/assets/3ba531af-b925-4573-aca6-5996be149f5b

https://github.com/user-attachments/assets/c9b60d2c-71df-49b1-a8cc-c55d423d6c71

### TESTING INSTRUCTIONS
1 - Check that the color scheme is applied when there are no color categories.
2 - Check that the color scheme is applied when there are color categories. 

### ADDITIONAL INFORMATION
- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/plugins/plugin-chart-echarts/src/Graph/transformProps.ts [hunk 1]
```diff
@@ -242,14 +248,39 @@ export default function transformProps(
...
+      );
+      sourceNodeColor = colorFn(sourceCategoryName);
+    }
+    if (targetCategory) {
+      targetCategoryName = getCategoryName(
+        targetCategory,
+        link[targetCategory],
+      );
+      targetNodeColor = colorFn(targetCategoryName);
+    }
```
Comment (justinpark) [inline superset-frontend/plugins/plugin-chart-echarts/src/Graph/transformProps.ts:270]: Can this let assignment consolidate the logic like this?
 
```suggestion
    const sourceCategoryName = sourceCategory ? getCategoryName(
        sourceCategory,
        link[sourceCategory],
      ) : undefined;
    const  targetCategoryName = targetCategory ? getCategoryName(
        targetCategory,
        link[targetCategory],
      ) : undefined;
    const sourceNodeColor = sourceCategory ? colorFn(sourceCategoryName) : colorFn('node');
    const targetNodeColor = targetCategory ? colorFn(targetCategoryName) : colorFn('node');
    const linkColor = colorFn('link');
```

File: superset-frontend/plugins/plugin-chart-echarts/src/Graph/transformProps.ts [hunk 2]
```diff
@@ -248,8 +254,26 @@ export default function transformProps(
...
       : undefined;
-    const sourceNode = getOrCreateNode(sourceName, source, sourceCategoryName);
-    const targetNode = getOrCreateNode(targetName, target, targetCategoryName);
+    const sourceNodeColor = sourceCategoryName
+      ? colorFn(sourceCategoryName)
+      : colorFn('node');
+    const targetNodeColor = targetCategoryName
+      ? colorFn(targetCategoryName)
+      : colorFn('node');
+    const linkColor = colorFn('link');
```
[the remark this answers is not in the document]
Comment (michael-s-molina) [reply superset-frontend/plugins/plugin-chart-echarts/src/Graph/transformProps.ts]: Make sense. Done.

File: superset-frontend/plugins/plugin-chart-echarts/src/Graph/transformProps.ts [hunk 3]
```diff
@@ -258,7 +282,9 @@ export default function transformProps(
       source: sourceNode.id,
       target: targetNode.id,
       value,
-      lineStyle: {},
+      lineStyle: {
+        color: firstColor,
```
Comment (villebro) [inline superset-frontend/plugins/plugin-chart-echarts/src/Graph/transformProps.ts:286]: Could we use `sourceNodeColor` here? I feel that helps distinguish the source/target better than just using `firstColor`

**Review by villebro — APPROVED** [review]
LGTM, thanks for the improvement and the multiple iterations!

**Pull request merged**